modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/KeyValueStorage.java

```diff
@@ -375,4 +375,11 @@ boolean invoke(
      * @see #saveCompactionRevision(long)
      */
     long getCompactionRevision();
+
+    /**
+     * Returns checksum corresponding to the revision, or {@code null} if the revision does not exist yet or was compacted.
+     *
+     * @param revision Revision.
+     */
+    @Nullable Long checksum(long revision);
```

Comment:
By analogy with `KeyValueStorage#timestampByRevision`, I would throw `CompactedException` for now.